flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java

```diff
@@ -38,11 +41,41 @@
  * @param <Message> The type of the message sent between vertices along the edges.
  * @param <EdgeValue> The type of the values that are associated with the edges.
  */
-public abstract class MessagingFunction<VertexKey extends Comparable<VertexKey> & Serializable, 
-	VertexValue extends Serializable, Message, EdgeValue extends Serializable> implements Serializable {
+public abstract class MessagingFunction<VertexKey extends Comparable<VertexKey> & Serializable,
+		VertexValue extends Serializable, Message, EdgeValue extends Serializable> implements Serializable {
 
 	private static final long serialVersionUID = 1L;
-	
+
+	// --------------------------------------------------------------------------------------------
+	//  Attributes that allow vertices to access their in/out degrees and the total number of vertices
+	//  inside an iteration.
+	// --------------------------------------------------------------------------------------------
+
+	private long numberOfVertices;
+
+	public long getNumberOfVertices() {
+		return numberOfVertices;
+	}
+
+	public void setNumberOfVertices(long numberOfVertices) {
```

Comment:
Normally it should be private. However, I need to access that setter in Graph's createVertexCentricIteration. If you can think of an alternative, let me know.